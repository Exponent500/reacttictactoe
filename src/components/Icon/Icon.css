.svg-container.white {
    color: white;
}