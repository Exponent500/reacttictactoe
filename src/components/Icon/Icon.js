import React from 'react';

import './Icon.css'

const Icon = ({ type, className }) => {
    return (
        <div className={"svg-container " + className}>
            <svg width='1em' height='1em'>
                <use xlinkHref={ `#${ type }` }></use>
            </svg>
        </div>
    )
}

export default Icon;