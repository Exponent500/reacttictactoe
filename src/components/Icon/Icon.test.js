import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Icon from './Icon';

let wrapped;

describe('Icon', () => {
    it('should render correctly', () => {
        wrapped = shallow(
            <Icon type='mockType'></Icon>
        )
        expect(shallowToJson(wrapped)).toMatchSnapshot();
    });
});