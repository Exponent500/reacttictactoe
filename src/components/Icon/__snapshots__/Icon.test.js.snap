// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<div
  className="svg-container undefined"
>
  <svg
    height="1em"
    width="1em"
  >
    <use
      xlinkHref="#mockType"
    />
  </svg>
</div>
`;
